### PR TITLE
Fix timeseries chart download link

### LINKED ISSUE #6112: Download chart data button - error message
It was noticed that the download data button has inconsistent behaviour for chart (able) datasets. 

**Working (i.e. CSV file created with charted dataset).** 
https://maps.dea.ga.gov.au/#share=s-blZFW2pswMZroPyMejRrfyPpQyj
image

Not working (i.e 404 error message shown)
https://nsw.digitaltwin.terria.io/#share=s-yliT5qzpSbAdZdfSfujUjNvddUc
image

image

If related to availability of data to be downloaded, the button should appear only when data is available or change the error message to clarify to the user "data is not available to download".

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

Fixes #6112 

This PR fixes the download link for the feature info panel chart for CSV items without a template strings.
  
### Test me

 - Open this [CI link](http://ci.terria.io/fix-timeseries-chart-download-link/#share=s-yZM68Po9YIfeybZhKmZwsS152bs&clean&https://gist.githubusercontent.com/na9da/4bfe951663bca8141a3ddd90c9f7ddf3/raw/6b95b4af35d90e9c04625d9e9fe6e3975adbe1c2/air-quality-liverpool.json).
 - Click any station point to open the feature info panel with a chart
 - Click the download button for the chart. It should trigger a CSV download.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated relevant documentation in `doc/`.
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/ReactViews/Custom/ChartCustomComponent.ts [hunk 1]
```diff
@@ -229,6 +241,15 @@ export default abstract class ChartCustomComponent<
...
 
+    // If downloads not specified but we have a body string, convert it to a downloadable data URI.
+    if (
+      attrs.downloads === undefined &&
+      body &&
+      this.constructDownloadUrlFromBody !== undefined
+    ) {
+      attrs.downloads = [this.constructDownloadUrlFromBody?.(body)];
+    }
+
```
Comment (na9da) [inline lib/ReactViews/Custom/ChartCustomComponent.ts:252]: When no download URLs are specified, make one from the chart data.

File: lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx [hunk 2]
```diff
@@ -825,21 +825,20 @@ function getTimeSeriesChartContext(catalogItem, feature, getChartDetails) {
...
+      const { title, csvData } = chartDetails;
       const result = {
         ...chartDetails,
         id: featureId?.replace(/\"/g, ""),
         data: csvData?.replace(/\\n/g, "\\n")
       };
       const idAttr = 'id="' + result.id + '" ';
-      const sourceAttr = 'sources="1"';
       const titleAttr = title ? `title="${title}"` : "";
-      result.chart = `<chart ${idAttr} ${sourceAttr} ${titleAttr}>${result.data}</chart>`;
```
Comment (na9da) [inline lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx:842]: There is no point passing `source="1"` as source. Instead now, when no source or download is specified, `ChartCustomComponent` will construct a downloadable link from the chart data.

File: test/ReactViews/Custom/Chart/ChartCustomComponentSpec.tsx [hunk 3]
```diff
@@ -14,12 +15,9 @@ import {
   DomElement,
   ProcessNodeContext
 } from "../../../../lib/ReactViews/Custom/CustomComponent";
-import UrlTraits from "../../../../lib/Traits/TraitsClasses/UrlTraits";
 import mixTraits from "../../../../lib/Traits/mixTraits";
 import MappableTraits from "../../../../lib/Traits/TraitsClasses/MappableTraits";
-
-const isComponentOfType: any = require("react-shallow-testutils")
-  .isComponentOfType;
+import UrlTraits from "../../../../lib/Traits/TraitsClasses/UrlTraits";
```
Comment (na9da) [inline test/ReactViews/Custom/Chart/ChartCustomComponentSpec.tsx:20]: Just import reorg.

**Review by nf-s — APPROVED** [review]
Love it!

**Pull request merged**